File: hash_table/0347_top_k_frequent_elements/0347_top_k_frequent_elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        hs = {}
        frq = {}
        for i in xrange(0, len(nums)):
            if nums[i] not in hs:
                hs[nums[i]] = 1
            else:
                hs[nums[i]] += 1
        
        for z,v in hs.iteritems():
            if v not in frq:
                frq[v] = [z]
            else:
                frq[v].append(z)
        
        arr = []

        for x in xrange(len(nums), 0, -1):
            if x in frq:
                for i in frq[x]:
                    arr.append(i)
        
        return [arr[x] for x in xrange(0, k)]

nums = [1,1,1,2,2,3]
k = 2

res = Solution().topKFrequent(nums, k)
print(res)

File: hash_table/0347_top_k_frequent_elements/0347_top_k_frequent_elements.go
package main 

import (
	"fmt"
	"sort"
)

func topKFrequent(nums []int, k int) []int {
	res := make([]int, 0, k)
	rec := make(map[int]int, len(nums))
	for _, n := range nums {
		rec[n]++
	}

	counts := make([]int, 0, len(rec))
	for _, c := range rec {
		counts = append(counts, c)
	}
	sort.Ints(counts)
	min := counts[len(counts)-k]

	for n, c := range rec {
		if c >= min {
			res = append(res, n)
		}
	}

	return res
}

func main() {
	nums := []int{1,1,1,2,2,3}
	k := 2
	res := topKFrequent(nums, k)
	fmt.Println(res)
}